gems/aws-crt/native/src/signable.c

```diff
@@ -0,0 +1,340 @@
+    struct aws_array_list *property_list = value;
+    aws_array_list_clean_up(property_list);
+}
+
+struct aws_signable *aws_crt_signable_new(void) {
+    struct aws_allocator *allocator = aws_crt_allocator();
+
+    struct aws_signable *signable = NULL;
+    struct aws_crt_signable_impl *impl = NULL;
+    aws_mem_acquire_many(
+        allocator, 2, &signable, sizeof(struct aws_signable), &impl, sizeof(struct aws_crt_signable_impl));
+
+    AWS_ZERO_STRUCT(*signable);
+    AWS_ZERO_STRUCT(*impl);
+
+    signable->allocator = allocator;
+    signable->vtable = &s_aws_crt_signable_vtable;
+    signable->impl = impl;
+
+    if (aws_hash_table_init(
+            &impl->properties,
+            allocator,
+            INITIAL_AWS_CRT_SIGNABLE_PROPERTIES_SIZE,
+            aws_hash_string,
+            aws_hash_callback_string_eq,
+            aws_hash_callback_string_destroy,
+            aws_hash_callback_string_destroy) ||
+        aws_hash_table_init(
+            &impl->property_lists,
+            allocator,
+            INITIAL_AWS_CRT_SIGNABLE_PROPERTY_LISTS_TABLE_SIZE,
+            aws_hash_string,
+            aws_hash_callback_string_eq,
+            aws_hash_callback_string_destroy,
+            s_aws_hash_callback_property_list_destroy) ||
+        aws_array_list_init_dynamic(
+            &impl->str_buffer,
+            allocator,
+            INITIAL_AWS_CRT_SIGNABLE_PROPERTY_LIST_SIZE * 2,
+            sizeof(struct aws_string *))) {
+        goto on_error;
+    }
+
+    return signable;
+
+on_error:
+
+    aws_signable_destroy(signable);
+
+    return NULL;
+}
+
+int aws_crt_signable_set_property(
+    struct aws_signable *signable,
+    const char *property_name,
+    const char *property_value) {
+
+    if (signable == NULL) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    struct aws_crt_signable_impl *impl = signable->impl;
+    if (impl == NULL) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    struct aws_string *name = NULL;
+    struct aws_string *value = NULL;
+
+    name = aws_string_new_from_c_str(signable->allocator, property_name);
+    value = aws_string_new_from_c_str(signable->allocator, property_value);
+
+    if (aws_hash_table_put(&impl->properties, name, value, NULL)) {
+        goto on_error;
+    }
+
+    return AWS_OP_SUCCESS;
+
+on_error:
+
+    aws_string_destroy(name);
+    aws_string_destroy(value);
+
+    return AWS_OP_ERR;
+}
+
+static struct aws_array_list *s_get_or_create_property_list(
+    struct aws_signable *signable,
+    const struct aws_string *list_name) {
+    struct aws_crt_signable_impl *impl = signable->impl;
+    struct aws_hash_element *element = NULL;
+
+    aws_hash_table_find(&impl->property_lists, list_name, &element);
+
+    if (element != NULL) {
+        return element->value;
+    }
+
+    struct aws_array_list *properties = aws_mem_acquire(signable->allocator, sizeof(struct aws_array_list));
+
+    AWS_ZERO_STRUCT(*properties);
```

Comment:
trivial: `aws_mem_calloc()` acquires AND zeroes out memory, prefer that over aws_mem_acquire() followed by AWS_ZERO_STRUCT